### PR TITLE
Feature std dev

### PR DESCRIPTION
**Pull request by arashahmadi564**
A new pull request to main branch

### CODE REVIEW

File: inflammation-analysis.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ def main(args):
     for filename in InFiles:
         inflammation_data = models.load_csv(filename)
 
-        view_data = {'average': models.daily_mean(inflammation_data), 'max': models.daily_max(inflammation_data), 'min': models.daily_min(inflammation_data)}
+        view_data = {'average': models.daily_mean(inflammation_data), 'max': models.daily_max(inflammation_data), 'min': models.daily_min(inflammation_data), **(models.s_dev(inflammation_data))}
```
Comment (Donnyvdm) [inline inflammation-analysis.py]: This line is too long, split over multiple rows for readability.
the **-syntax is confusing and inconsistent, I would prefer explicit names. These changes depend on changes in models.py

```suggestion
        view_data = {
            'average': models.daily_mean(inflammation_data), 
            'max': models.daily_max(inflammation_data), 
            'min': models.daily_min(inflammation_data), 
            'stddev': models.daily_standard_deviation(inflammation_data))
        }
```

File: inflammation/models.py [hunk 2]
```diff
@@ -32,3 +32,13 @@ def daily_min(data):
     """Calculate the daily min of a 2d inflammation data array."""
     return np.min(data, axis=0)
 
+
+def s_dev(data):
```
Comment (Donnyvdm) [inline inflammation/models.py]: Rename to a more explicit name
```suggestion
def daily_standard_deviation(data):
```

File: inflammation/models.py [hunk 3]
```diff
@@ -32,3 +32,13 @@ def daily_min(data):
...
+    """Computes and returns standard deviation for data."""
```
Comment (Donnyvdm) [inline inflammation/models.py]: Explain the input data / arguments in the docstring.

File: inflammation/models.py [hunk 4]
```diff
@@ -32,3 +32,13 @@ def daily_min(data):
...
+    mmm = np.mean(data, axis=0)
+    devs = []
+    for entry in data:
+        devs.append((entry - mmm) * (entry - mmm))
+
+    s_dev2 = sum(devs) / len(data)
+    return {'standard deviation': s_dev2}
```
Comment (Donnyvdm) [inline inflammation/models.py]: All other functions in the model return a numpy array, while this function returns a dictionary. I think it's better to stay consistent
```suggestion
    return std_devs
```

File: inflammation/models.py [hunk 5]
```diff
@@ -32,3 +32,13 @@ def daily_min(data):
...
 
+
+def s_dev(data):
+    """Computes and returns standard deviation for data."""
+    mmm = np.mean(data, axis=0)
+    devs = []
+    for entry in data:
+        devs.append((entry - mmm) * (entry - mmm))
+
+    s_dev2 = sum(devs) / len(data)
```
Comment (Donnyvdm) [inline inflammation/models.py]: Rename variable to be more explicit
```suggestion
    std_devs = sum(devs) / len(data)
```

Comment (Donnyvdm) [inline inflammation/models.py]: The standard deviation is calculated explictly, but I believe numpy has a built-in function for that np.std
```suggestion
   std_devs = np.std(data, axis=0)
```

File: tests/test_models.py [hunk 6]
```diff
@@ -29,3 +31,13 @@ def test_daily_mean_integers():
     # Need to use Numpy testing functions to compare arrays
     npt.assert_array_equal(daily_mean(test_input), test_result)
 
+
+@pytest.mark.parametrize('data, expected_standard_deviation', [
```
Comment (Donnyvdm) [inline tests/test_models.py:35]: Add test cases for multiple patients
Add test a test case for patients with a single measurement
Add a test case with values that are closer to the real data (up to 100)

**Review by Donnyvdm — CHANGES_REQUESTED** [review]

File: tests/test_models.py (hunk 6 above)
Comment (arashahmadi564) [reply tests/test_models.py:35]: will do!

File: inflammation/models.py (hunk 3 above)
Comment (arashahmadi564) [reply inflammation/models.py]: done

**Pull request merged**